base/compiler/ssair/irinterp.jl

```diff
@@ -94,16 +94,24 @@ function getindex(tpdum::TwoPhaseDefUseMap, idx::Int)
     return TwoPhaseVectorView(tpdum.data, nelems, range)
 end
 
-struct IRInterpretationState
+# TODO add `result::InferenceResult` & `parent::InferenceState` for this
+struct IRInterpretationState <: AbsIntState
+    src::CodeInfo
```

Comment:
I would prefer not to have to have a `CodeInfo` in here? What information are we missing from the IRCode?

Reply:
Yes, but we will need `src.method_for_recursion_heuristic` etc.

Reply:
I'd prefer to separate that out into its own minimal struct rather than dumping the whole CodeInfo in here. For non-Base use cases, we may not have a CodeInfo at all.

Reply:
Yeah I understand. I will work on this.